### ISSUE TITLE
CSV Fails to load. Eventual conflict with Parsers dependency

### ISSUE THREAD

**Issue by sunbubble** [JuliaData/CSV.jl#304]
Julia v0.7
I'm not using the dev version.
I've tried Pkg.resolve() and also reproduced the same situation in Linux and macOs

```Julia
julia> using CSV
[ Info: Precompiling CSV [336ed68f-0bac-5ca0-87d4-7b16caf5d00b]
ERROR: LoadError: ArgumentError: Package CSV does not have Parsers in its dependencies:
- If you have CSV checked out for development and have
  added Parsers as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with CSV
Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:830
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1038
 [4] include(::Module, ::String) at ./sysimg.jl:29
 [5] top-level scope at none:2
 [6] eval at ./boot.jl:319 [inlined]
 [7] eval(::Expr) at ./client.jl:399
 [8] top-level scope at ./none:3
in expression starting at /Users/sunbubble/.julia/packages/CSV/Jf3xI/src/CSV.jl:3
ERROR: Failed to precompile CSV [336ed68f-0bac-5ca0-87d4-7b16caf5d00b] to /Users/sunbubble/.julia/compiled/v0.7/CSV/HHBkp.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1185
 [4] _require(::Base.PkgId) at ./logging.jl:311
 [5] require(::Base.PkgId) at ./loading.jl:852
 [6] macro expansion at ./logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:834
```

**Comment by sunbubble** [comment]
I've got it working by running Pkd.add("Parsers") but now I always get
```Julia
julia> using CSV
[ Info: Precompiling CSV [336ed68f-0bac-5ca0-87d4-7b16caf5d00b]
┌ Warning: Package CSV does not have Parsers in its dependencies:
│ - If you have CSV checked out for development and have
│   added Parsers as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with CSV
└ Loading Parsers into CSV from project dependency, future warnings for CSV are suppressed.
```
the warnings aren't suppressed.

**Comment by quinnj** [comment]
Fixed via 5009d931dce59c499ad0bf839f4df485615bfb90; just tagged a new release.
